.github/workflows/gitops_canary.yaml

```diff
@@ -0,0 +1,189 @@
+
+    - name: Build and push
+      uses: docker/build-push-action@v3
+      with:
+        driver: docker
+        file: deploy/kubernetes/docker/Dockerfile
+        platforms: ${{ inputs.docker-platforms }}
+        push: true
+        tags: |
+          ${{ steps.variables.outputs.repo }}:${{ steps.variables.outputs.tag }}
+          ${{ steps.docker-login.outputs.ecr-registry }}/${{ steps.variables.outputs.repo }}:${{ steps.variables.outputs.tag }}
+        build-args: |
+          VERSION=${{ steps.variables.outputs.tag }}
+        secrets: |
+          "github_token=${{ secrets.GITHUB_REPOS_PAT }}"
+
+    - name: Gitops
+      uses: ovatu/actions/.github/actions/gitops-release@main
+      with:
+        name: ${{ inputs.gitops-name }}-release
+        repository: ${{ inputs.gitops-repo }}
+        repository-ssh-key: ${{ secrets.GITOPS_REPO_PRIVATE_KEY }}
+        input-directory: ${{ inputs.helm-chart }}
+        output-directory: apps/${{ inputs.gitops-name }}/release
+        message: "[Actions][Build] repo:${{ inputs.gitops-name }}, canary: ${{ steps.variables.outputs.release }}"
+        values-script: |
+          echo "BEGIN writing values"
+          yq -i '.appVersion = "${{ steps.variables.outputs.tag }}"' Chart.yaml
+          yq -i '.image.repository = "${{ steps.docker-login.outputs.ecr-registry }}/${{ steps.variables.outputs.repo }}"' values.yaml
+          yq -i '.image.tag = "${{ steps.variables.outputs.tag }}"' values.yaml
+          echo "DONE writing values"
+          
+          export CHART_NAME=$(yq '.name' ../chart/Chart.yaml)
+          export CHART_VERSION=$(yq '.version' ../chart/Chart.yaml)
+          
+          yq -i ".name = \"${{ inputs.gitops-name }}-${{ steps.variables.outputs.slug }}\"" Chart.yaml
+          yq -i ".version = strenv(CHART_VERSION)" Chart.yaml
+          yq -i ".dependencies[0].name = strenv(CHART_NAME)" Chart.yaml
+          yq -i ".dependencies[0].version = strenv(CHART_VERSION)" Chart.yaml
+          yq -i ".dependencies[0].repository = \"file://../chart\"" Chart.yaml
+          yq -i ".dependencies[0].alias = \"canary\"" Chart.yaml
+          
+          rsync -vr --delete $GITHUB_ACTION_PATH/templates/ templates/
+          
+          yq -i ".namespace.enabled = true" values.yaml
+          yq -i ".namespace.name = \"${{ inputs.gitops-name }}-${{ steps.variables.outputs.slug }}\"" values.yaml
+
+          echo "BEGIN inputs.gitops-values-script"
+          ${{ inputs.gitops-values-script }}
+          echo "END inputs.gitops-values-script"
+          
+          echo "BEGIN inputs.gitops-canary-values-script"
+          ${{ inputs.gitops-canary-values-script }}
+          echo "END inputs.gitops-canary-values-script"
+
+    - name: Send Slack webhook
+      uses: slackapi/slack-github-action@v1.24.0
+      with:
+        payload: |
+          {
+            "text": "Released: ${{ inputs.name }}, v: ${{ steps.variables.outputs.release }}",
```

Comment:
Maybe a little more 'canary related'

Reply:
Sounds good, and I changed it to the devops slack channel which probably makes more sense for canaries